openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -1810,10 +1810,12 @@ struct AMDGPUDeviceTy : public GenericDeviceTy, AMDGenericDeviceTy {
       return Err;
     GridValues.GV_Warp_Size = WavefrontSize;
 
-    // Get the frequency of the steady clock.
-    if (auto Err = getDeviceAttr(HSA_AMD_AGENT_INFO_TIMESTAMP_FREQUENCY,
-                                 ClockFrequency))
-      return Err;
+    // Get the frequency of the steady clock. If the attribute is missing
+    // assume running on an older libhsa and default to 0, omp_get_wtime
+    // will be inaccurate but otherwise programs can still run.
+    if (auto Err = getDeviceAttrRaw(HSA_AMD_AGENT_INFO_TIMESTAMP_FREQUENCY,
```

Comment:
So `Raw` does not emit an error. I thought the issue would be that `HSA_AMD_AGENT_INFO_TIMESTAMP_FREQUENCY` may not be defined at all so it would fail to compile?

Reply:
If the enum field is missing, no compilation, no progress. I don't think there's a reasonable way to reflect on whether an enum field is present or not (i.e. no `#ifdef` equivalent).

However the enum field is present in the dynamic_hsa header, so openmp compiles just fine against that.

Reply:
Was assuming there'd be some version field in HSA that indicates it, but I don't know. This should be better than nothing for now.

Reply:
There is no version in the HSA library, it's on the wishlist. There is one in the header, but as the header may bear little relation to the shared library, that is a partial help.